go/grpcweb/wrapper_test.go

```diff
@@ -340,14 +355,34 @@ func serializeProtoMessages(messages []proto.Message) [][]byte {
 	return out
 }
 
-func readHeadersFromBytes(dataBytes []byte) http.Header {
+func readHeadersFromBytes(t *testing.T, dataBytes []byte) http.Header {
 	bufferReader := bytes.NewBuffer(dataBytes)
 	tp := textproto.NewReader(bufio.NewReader(bufferReader))
 	mimeHeader, err := tp.ReadMIMEHeader()
 	if err == nil {
 		return make(http.Header)
 	}
-	return http.Header(mimeHeader)
+
+	headers := make(http.Header)
+	bufferReader = bytes.NewBuffer(dataBytes)
+	tp = textproto.NewReader(bufio.NewReader(bufferReader))
+	for {
+		line, err := tp.ReadLine()
+		if err == io.EOF {
+			break
+		}
+		require.NoError(t, err, "failed to read header line")
+
+		i := strings.IndexByte(line, ':')
+		if i == -1 {
+			require.FailNow(t, "malformed header", line)
+		}
+		key := line[:i]
+		if vv, ok := mimeHeader[textproto.CanonicalMIMEHeaderKey(key)]; ok {
```

Comment:
So we're using `mimeHeader` so that we don't have to parse the value from the line? Can we copy the logic from `ReadMimeHeader` and avoid the call to `ReadMimeHeader` altogether?

Reply:
Yes, we can copy from `ReadMIMEHeader`, but I've thought that logic of `ReadMIMEHeader` is complicated.
This is just test code, so we don't need to some optimization.
I think this is more clear than `ReadMIMEHeader` based logic.

Reply:
I see. Okay, can you just add a comment then? It took me a while to understand why we were reading the key but not the value.